controller/src/controller.rs

```diff
@@ -261,6 +272,28 @@ fn sort_shadows(shadows: &mut Vec<BottlerocketShadow>, associated_brs_name: &str
     }
 }
 
+/// Fetch the environment variable to determine the max concurrent update nodes number.
+fn get_max_concurrent_update() -> Result<usize> {
+    let max_concurrent_update = env::var(MAX_CONCURRENT_UPDATE_ENV_VAR)
+        .context(controllerclient_error::MissingEnvVariable {
+            variable: MAX_CONCURRENT_UPDATE_ENV_VAR.to_string(),
+        })?
+        .to_lowercase();
+
+    if max_concurrent_update.eq("unlimited") {
+        return Ok(usize::MAX);
+    } else {
+        max_concurrent_update
+            .parse::<usize>()
+            .context(controllerclient_error::MaxConcurrentUpdateParseError)
+    }
+}
+
+/// Determine if a BottlerocketShadow is in default or None status.
+fn is_initial_state(brs: &BottlerocketShadow) -> bool {
+    brs.status == None
+        || brs.status.clone().unwrap().current_state == BottlerocketShadowState::default()
```

Comment:
I think a more "rusty" way to write that is:

```suggestion
   match brs.status {
       Some(status) => status.current_state == BottlerocketShadowShadowState::default(),
       None => false
   }
```

Reply:
I adjusted a little based on the original logic and the ownership.